apis/flowcollector/v1beta1/flowcollector_types.go

```diff
@@ -716,6 +724,64 @@ type FlowCollectorLoki struct {
 	StatusTLS ClientTLS `json:"statusTls"`
 }
 
+// `PrometheusQuerierManual` defines the full connection parameters to Prometheus.
+type PrometheusQuerierManual struct {
+	//+kubebuilder:default:="http://prometheus:9090"
+	// `url` is the address of an existing Prometheus service to use for querying metrics.
+	URL string `json:"url,omitempty"`
+
+	// TLS client configuration for Prometheus URL.
+	// +optional
+	TLS ClientTLS `json:"tls"`
+
+	// Set `true` to forward logged in user token in queries to Prometheus
+	// +optional
+	ForwardUserToken bool `json:"forwardUserToken"`
+}
+
+type PrometheusMode string
+
+const (
+	PromModeAuto   PrometheusMode = "Auto"
+	PromModeManual PrometheusMode = "Manual"
+)
+
+// `FlowCollectorPrometheus` defines the desired state for usage of Prometheus.
+type FlowCollectorPrometheus struct {
+	// Prometheus querying configuration, such as client settings, used in the Console plugin.
+	Querier PrometheusQuerier `json:"querier,omitempty"`
+}
+
+// `PrometheusQuerier` defines the desired state for querying Prometheus (client...)
+type PrometheusQuerier struct {
+	// Set `enable` to `true` to make the Console plugin querying flow metrics from Prometheus instead of Loki whenever possible.
+	// The Console plugin can use either Loki or Prometheus as a data source for metrics (see also `spec.loki`), or both.
+	// Not all queries are transposable from Loki to Prometheus. Hence, if Loki is disabled, some features of the plugin are disabled as well,
+	// such as getting per-pod information or viewing raw flows.
+	// If both Prometheus and Loki are enabled, Prometheus takes precedence and Loki is used as a fallback for queries that Prometheus cannot handle.
+	// If they are both disabled, the Console plugin is not deployed.
+	//+kubebuilder:default:=true
```

Comment:
// +optional

Reply:
fields are optional by default (you can make the test, marking this field as optional won't do anything when regenerating the crd openapi) - see also https://book.kubebuilder.io/reference/markers/crd-validation.html
=>
> // +optional
> specifies that this field is optional, **if fields are required by default**.